app/services/LanguageToolFactory.scala

```diff
@@ -53,10 +52,12 @@ class LanguageToolFactory(
 }
 
 class LanguageTool(category: String, instance: JLanguageTool)(implicit ec: ExecutionContext) extends Validator {
+  def getId = "language-tool"
+
   def getCategory = category
 
-  def check(request: ValidatorRequest): List[RuleMatch] = {
-    instance.check(request.text).asScala.map(RuleMatch.fromLT).toList
+  def check(request: ValidatorRequest): Future[List[RuleMatch]] = {
+    Future.successful(instance.check(request.text).asScala.map(RuleMatch.fromLT).toList)
```

Comment:
I think this will execute `instance.check` synchronously on the current thread vs `Future { instance.check(...) }` which will be asynchronous in the `ExecutionContext`.

Not sure which one we want here, just flagging in case

Reply:
Good question, and I'm not sure either -- let's chat 👍

Reply:
You were correct, and stream concurrency was not a thing as a result. It's now resolved 👍